ntp-proto/src/filter.rs

```diff
@@ -58,6 +61,52 @@ impl FilterTuple {
     fn is_dummy(self) -> bool {
         self == Self::DUMMY
     }
+
+    /// The default logic for updating a peer with a new packet.
+    ///
+    /// A Broadcast association requires different logic.
+    /// All other associations should use this function
+    #[allow(dead_code)]
+    fn from_packet_default(
+        packet: &NtpHeader,
+        system_precision: NtpDuration,
+        destination_timestamp: NtpTimestamp,
+        local_clock_time: NtpTimestamp,
+    ) -> Self {
+        // for reference
+        //
+        // | org       | T1         | origin timestamp      |
+        // | rec       | T2         | receive timestamp     |
+        // | xmt       | T3         | transmit timestamp    |
+        // | dst       | T4         | destination timestamp |
+
+        // for a broadcast association, different logic is used
+        debug_assert_ne!(packet.mode, NtpAssociationMode::Broadcast);
+
+        let packet_precision = NtpDuration::from_exponent(packet.precision);
+
+        // offset is the average of the deltas (T2 - T1) and (T4 - T3)
+        let offset1 = packet.receive_timestamp - packet.origin_timestamp;
```

Comment:
We should probably not be using origin timestamp from the packet, but instead keep our sending time locally and using that. This would allow us to randomize origin_timestamp more in the future to make attacks harder, and to use more precise timestamping once we have kernel timestamping working.

Reply:
we could update the `origin_timestamp` field of the packet. That would be less error-prone (the packet is stored in the peer, so it's very easy to (mis)use the packet's origin timestamp in the logic.) Thoughts?

Reply:
Fair point, that seems reasonable.